client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -135,6 +136,12 @@ export default function getThankYouPageUrl( {
 	if ( redirectTo ) {
 		const { protocol, hostname, port, pathname, searchParams } = getUrlParts( redirectTo );
 
+		// We need to do a hard redirect if we're redirecting to the stepper.
+		// Since stepper is self-contained, it doesn't load properly if we do a normal history state change
+		if ( isURL( redirectTo ) || redirectTo.startsWith( '/setup' ) ) {
+			window.location.href = redirectTo;
```

Comment:
This function is covered by unit tests to prevent regressions. Could you add a unit test for this new behavior in `client/my-sites/checkout/get-thank-you-page-url/test/get-thank-you-page-url.ts`?

Also... 😬  is actually _performing_ a redirect in a function called `getThankYouPageUrl()` really a good idea? That seems like a very slippery slope for those who use this function (eg: it's called several times in checkout before a transaction is complete). Could this be moved to the code that actually performs the redirect instead?